### PR TITLE
clean up error_on_nested_jit_trace flag

### PR DESCRIPTION
**Pull request by bobrenjc93**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #137309
* __->__ #136961

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames @rec

### CODE REVIEW

File: torch/_dynamo/config.py [hunk 1]
```diff
@@ -334,10 +334,6 @@ def _get_optimize_ddp_mode():
 # If True, raise when aot autograd is unsafe to use
 raise_on_unsafe_aot_autograd = False
 
-# If true, error if you torch.jit.trace over a dynamo-optimized function.
-# If false, silently suppress dynamo
-error_on_nested_jit_trace = True
```
Comment (anijain2305) [inline torch/_dynamo/config.py:339]: Dont delete the flag from here.

**Review by anijain2305 — DISMISSED** [review]

**Review by jansel — DISMISSED** [review]
Please add a comment to `config.py` that `error_on_nested_jit_trace` is ignored and maintained for backwards compatbility.

Also do this for all the other ignored flags.

**Comment by bobrenjc93** [comment]
> Please add a comment to `config.py` that `error_on_nested_jit_trace` is ignored and maintained for backwards compatbility.
> 
> Also do this for all the other ignored flags.

Done. I only landed one other PR without this comment. For that PR I made another PR adding the comment https://github.com/pytorch/pytorch/pull/137309

**Review by jansel — APPROVED** [review]

**Pull request closed without merging**